scripts/estimate-builds.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python3
+# pylint: disable=invalid-name
+
+import datetime
+from pathlib import Path
+import sys
+
+import croniter
+
+# Add the root of the repo to PYTHONPATH for utils
+ci_root = Path(__file__).resolve().parents[1]
+sys.path.insert(0, str(ci_root))
+# pylint: disable-next=wrong-import-position
+from utils import get_config_from_generator  # noqa: E402
+
+config = get_config_from_generator()
+
+now = datetime.datetime.now(tz=datetime.timezone.utc)
+week_from_now = now + datetime.timedelta(weeks=1)
+
+builds_per_tree = {}
+for tree in config['tree_schedules']:
+    tree_name = tree['name']
+    tree_llvm_ver = tree['llvm_version']
+
+    # Calculate the number of times that a workflow runs in a week based on its
+    # schedule
+    num_runs = len(
+        list(croniter.croniter_range(now, week_from_now, tree['schedule'])))
```

Comment:
Though we have many different builds, we only have a small number of schedules. Recomputing this number for each build is expensive. Consider throwing the result in a dict keyed on `tree['schedule']`.

Reply:
oh! Indeed not worth it!

Reply:
From profiling `python3 -m cProfile -s cumtime ./scripts/estimate-builds.py | less` it looks like runtime is dominated by parsing the generator `utils.py:9(get_config_from_generator)`.